### DISCUSSION TITLE
Try to run tcp-client example

### DISCUSSION THREAD

**Discussion by MichelAraujo** [libuv/libuv#4265] [Q&A]
Hello!
I was tried run this example about tcp client:
https://github.com/delaemon/libuv-tutorial/blob/master/tcp-client.c
Its compiled but not return anything whan I run, I tried put a printf in "on_connect" function e don't show anything, some adivice?

**Comment by santigimeno** [comment]
Yes, the loop is not being run. Adding
```
return uv_run(loop, UV_RUN_DEFAULT);
```
at the end of the main function should do it. There are other issues in that code though (for example, when checking the `status` in the `connect_cb`)

**Reply by MichelAraujo** [reply]
I added the "uv_run" and the print inside the "on_connect" worked, now the callback of the "uv_read_start" that is not called, not show the print.
I tryed access the google page from IP

**Reply by MichelAraujo** [reply]
after the "uv_tcp_connect" I have to make more something to read the data? To to make "uv_read_cb" run?  I realise when I add an invalid port the "uv_read_cb" is called, show the printf.

**Reply by MichelAraujo** [reply]
It worked!!
I had to set the http request in "uv_write" message buffer like:
```
    uv_buf_t buf = uv_buf_init(buffer, sizeof(buffer));
    char *message = (char*)"GET / HTTP/1.1\r\nHost: localhost\r\nConnection: close\r\n\r\n";
    buf.len = strlen(message);
    buf.base = message;
```
I'll try write a tutorial more clear for the community after that.
Thanks for your time!
